protocol.html

```diff
@@ -717,6 +719,62 @@ <h4 property="schema:name"><span>Note</span>: Conditional Update</h4>
+                      <li id="server-patch-n3-formulae" rel="spec:requirement" resource="#server-patch-n3-formulae"><span property="spec:statement">When present, <code>?deletions</code>, <code>?insertions</code>, and <code>?conditions</code> <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> be non-nested <em>cited formulae</em> [<cite><a class="bibref" href="#bib-notation3">N3</a></cite>] consisting only of triples and/or triple patterns [<cite><a class="bibref" href="#bib-sparql11-query">SPARQL11-QUERY</a></cite>]. When not present, they are presumed to be the empty formula <code>{}</code>.</span></li>
+                    </ul>
+
+                    <p>While other specifications might provide a structure and interpretation for a wider class of N3 Patch documents, the present specification only governs the application of N3 Patch documents that additionally adhere to the following structural constraints:</p>
+
+                    <ul>
+                      <li id="server-patch-n3-single" rel="spec:requirement" resource="#server-patch-n3-single"><span property="spec:statement">The patch document <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> contain exactly one patch resource, identified by one or more of the triple patterns described above, which all share the same <code>?patch</code> subject.</span></li>
+                      <li id="server-patch-n3-matching-strategy" rel="spec:requirement" resource="#server-patch-n3-matching-strategy"><span property="spec:statement">The patch resource <span rel="spec:requirementLevel" resource="spec:MAY">MAY</span> optionally contain one triple of the form <code>?patch solid:matchingStrategy solid:SingleMatch</code>.</span></li>
+                      <li id="server-patch-n3-variables" rel="spec:requirement" resource="#server-patch-n3-variables"><span property="spec:statement">The <code>?insertions</code> and <code>?deletions</code> formulae <span rel="spec:requirementLevel" resource="spec:MUSTNOT">MUST NOT</span> contain variables that do not occur in the <code>?conditions</code> formula.</span></li>
+                      <li id="server-patch-n3-blank-nodes" rel="spec:requirement" resource="#server-patch-n3-blank-nodes"><span property="spec:statement">The <code>?insertions</code> and <code>?deletions</code> formulae <span rel="spec:requirementLevel" resource="spec:MUSTNOT">MUST NOT</span> contain blank nodes.</span></li>
+                      <li id="server-patch-n3-other-triples" rel="spec:requirement" resource="#server-patch-n3-identifier"><span property="spec:statement">The patch document <span rel="spec:requirementLevel" resource="spec:MUSTNOT">MUST NOT</span> contain any other triples.</span></li>
```

Comment:
It is necessary for forward compatibility: what if patch functionality is altered by such statements? Please advise.

Reply:
I don't have a hard stance on this. Just want to make sure we cover what we really want out of this.

---

Perhaps I'm not understanding how other statements can modify the patch functionality. Isn't that out of spec behaviour? Or do you mean stuff getting added that's unspecified.. and later on a spec comes up and uses a pattern which may or may not conflict with existing N3 Patch documents? (New extensions shouldn't spec stuff that conflicts with existing use) Or do you mean another specification could make it so? My understanding is that only what's specified will be processed and the rest will be ignored. That aside, it is an RDF document.

---

Another way to look at this may from the point of extensions basically saying that new stuff is allowed as long as they don't contradict (fudge up) what's defined in this spec. Perhaps a much shorter version of the following will cover (from the Notifications Protocol / WAC):

>Extensions incorporate additional features beyond what is defined in this specification. Extensions MUST NOT contradict nor cause the non-conformance of functionality defined in this specification.

Not necessarily in terms of "extension" but along those lines. Only look at solid:Patch stuff..

Reply:
> Or do you mean stuff getting added that's unspecified.. and later on a spec comes up

Exactly; see https://github.com/solid/specification/pull/346#discussion_r758183032

> (New extensions shouldn't spec stuff that conflicts with existing use)

Agreed.

> Or do you mean another specification could make it so?

Yeah.

Example:

```
_:rename a solid:Patch;
   ex:condition { ?person ex:startWith "http://example.org" };
   solid:where   { ?person ex:familyName "Anchor". };
   solid:inserts { ?person ex:givenName "An". };
   solid:deletes { ?person ex:givenName "Bob". }.
```

where `ex:condition` adds a stronger constraint.

> Another way to look at this may from the point of extensions basically saying that new stuff is allowed as long as they don't contradict (fudge up) what's defined in this spec.

It does not, because of https://github.com/solid/specification/pull/346#discussion_r758183032, where we say that this spec only governs very specify patch documents.

Happy to consider other solutions; for instance, mandating another `rdf:type` in such cases.

Reply:
They're essentially the same. It entails that extensions can be subclassed.

Can no other statement be part of this document e.g. `_:rename rdfs:label "...".`, `<> dcterms:creator <...>.` ?

Reply:
I agree it is important to remove the requirement that there are no other triples.  That's not the way we work with RDF.  We wok by ignoring triples which we don't understand.   And when we add triples, we make designs which will be fine if the triples are ignored.   (See the discussion on whether the Patch rdf:type is reuired, which allows us to make new things with different semantics).

Reply:
I agree. I initially wanted to take the shortcut of allowing no other triples for 0.9, but we can bypass this if we introduce a specific type. Resuming in https://github.com/solid/specification/pull/346#discussion_r768611736